rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -261,21 +274,42 @@ void gs_frame::keyPressEvent(QKeyEvent *keyEvent)
 		}
 		break;
 	case Qt::Key_R:
-		if (keyEvent->modifiers() == Qt::ControlModifier && !m_disable_kb_hotkeys)
+		if (keyEvent->modifiers() == Qt::ControlModifier && !m_disable_kb_hotkeys && !g_cfg.savestate.suspend_emu && !Emu.GetTitleID().empty() && (Emu.IsRunning() || Emu.GetStatus() == system_state::paused))
 		{
-			switch (Emu.GetStatus())
-			{
-			case system_state::ready:
+			extern bool is_savestate_compatible(const fs::file& file);
+
+			const std::string save_dir = fs::get_cache_dir() + "/savestates/";\
+
+			std::string savestate_path;
+			u64 mtime = 0;
+
+			for (auto&& entry : fs::dir(save_dir))
```

Comment:
Why not do this in the new utils file?

Reply:
It is a very specific situation and is not directly related to savestates. (picking up the most recent file that is)

Reply:
A random directory scanner shouldn't be in a game window class

Reply:
Everything after m_disable_kb_hotkeys should be in a savestate function